source/MRMesh/MRMatrix2.h

```diff
@@ -51,18 +51,24 @@ struct Matrix2
     constexpr T trace() const noexcept { return x.x + y.y; }
     /// compute sum of squared matrix elements
     constexpr T normSq() const noexcept { return x.lengthSq() + y.lengthSq(); }
-    constexpr T norm() const noexcept { return std::sqrt( normSq() ); }
+    constexpr auto norm() const noexcept
+    {
+        // Calling `sqrt` this way to hopefully support boost.multiprecision numbers.
+        // Returning `auto` to not break on integral types.
```

Comment:
for integral types `sqrt` is not defined

Reply:
It automatically casts them to double.